tracer/src/Datadog.Trace/FeatureFlags/FeatureFlagsModule.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="FeatureFlagsModule.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Exposure;
+using Datadog.Trace.FeatureFlags.Exposure.Model;
+using Datadog.Trace.FeatureFlags.Rcm;
+using Datadog.Trace.FeatureFlags.Rcm.Model;
+using Datadog.Trace.Logging;
+using Datadog.Trace.RemoteConfigurationManagement;
+
+namespace Datadog.Trace.FeatureFlags
+{
+    internal sealed class FeatureFlagsModule : IDisposable
+    {
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(FeatureFlagsModule));
+
+        private readonly IRcmSubscriptionManager _rcmSubscriptionManager;
+        private ISubscription? _rcmSubscription = null;
+        private FfeProduct? _ffeProduct = null;
+        private Action? _onNewConfigEventHander = null;
+        private ExposureApi? _exposureApi = null;
+
+        private FeatureFlagsEvaluator? _evaluator = null;
+
+        internal FeatureFlagsModule(TracerSettings settings, IRcmSubscriptionManager? rcmSubscriptionManager = null)
+        {
+            Log.Debug("FeatureFlagsModule ENABLED");
+            _rcmSubscriptionManager = rcmSubscriptionManager ?? RcmSubscriptionManager.Instance;
+            _exposureApi = ExposureApi.Create(settings);
+            _ffeProduct = new FfeProduct(UpdateRemoteConfig);
+            if (Interlocked.Exchange(ref _rcmSubscription, new Subscription(_ffeProduct.UpdateFromRcm, RcmProducts.FfeFlags)) == null)
+            {
+                _rcmSubscriptionManager.SubscribeToChanges(_rcmSubscription!);
+                _rcmSubscriptionManager.SetCapability(RcmCapabilitiesIndices.FfeFlagConfigurationRules, true);
+            }
+        }
+
+        public static FeatureFlagsModule? Create(TracerSettings settings, IRcmSubscriptionManager? rcmSubscriptionManager = null)
+        {
+            if (settings.IsFlaggingProviderEnabled)
+            {
+                return new FeatureFlagsModule(settings, rcmSubscriptionManager ?? RcmSubscriptionManager.Instance);
+            }
+
+            return null;
+        }
```

Comment:
nit: if RCM manager is always required, lets not introduce "hidden" usages of global statics, and instead force passing it in instead.


```suggestion
        public static FeatureFlagsModule? Create(TracerSettings settings, IRcmSubscriptionManager? rcmSubscriptionManager)
        {
            if (settings.IsFlaggingProviderEnabled)
            {
                return new FeatureFlagsModule(settings, rcmSubscriptionManager);
            }

            return null;
        }
```

Reply:
Rremoved